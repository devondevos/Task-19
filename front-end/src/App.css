* {
  padding: 0px;
  margin: 0px;
}

body {
  background-image: url("apple.jpg");
  background-size: 100%;
  background-repeat: no-repeat;
}

header {
  background-image: url("header.jpg");
  background-size: 100%;
  background-position: center;
  background-repeat: no-repeat;
  height: 220px;
}

#favback, #music, #bookdiv {
  background-color: white;
}

nav {
  text-align: right;
}

.Songinput {
  width: 150px;
  transition: 1000ms;
  height: 20px;
}

.track-info, .Favinfo, .Bookinfo {
  padding-left: 70px;
}

#favback {
  padding-left: 70px;
}

.searchItem {
  text-align: center;
}

.Songinput:hover {
  width: 300px;
}

.Searchbutton {
  width: 70px;
  height: 30px;
  background-color: lightgrey;
  border: 0.5px solid black;
}

.Searchbutton:hover {
  background-color: black;
  color: white;
}

li, a {
  list-style-type: none;
  text-decoration: none;
  color: black;
  display: inline-block;
  padding: 7px;
}

li:hover {
  background-color: white;
  border-radius: 5px;
}

nav {
  background-color: lightgrey;
}