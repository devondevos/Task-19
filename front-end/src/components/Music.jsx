// Import react
import React from 'react';

class Music extends React.Component{
    constructor(){
        super()
        this.state ={
            data: [],
            musicInput: ''
        }
    }
    Searchnew(){
        fetch('/music')
            .then(res => res.json())
            .then(data => this.setState({data}, () => console.log('info is fetched...', data)))
    }
    Searchmusic = async () => {
        let search = this.state.musicInput.split(' ').join('+')
        const getMusic = await fetch(`/music?search=${search}&type=${this.state.type}`)
        let res = await getMusic.json()
        this.setState({
            data: res
        })
    }

    favouriteMusic= (i) => {
        let favPic = {
            id: i.trackId,
            artist: i.artistName,
            artwork: i.artworkUrl100,
            track: i.trackName,
            sample: i.previewUrl
        }
        fetch('/favoritesMusic', {
            method: 'POST',
            headers : {
                "Content-Type" : "application/json"
            },
            body: JSON.stringify(favPic)
        })
    }
    render(){
        return (
            <div id="music">
                <div class="searchItem">
                    <input class="Songinput" type="text" placeholder="Search Tracks" onChange={(e) => this.setState({musicInput: e.target.value})}/>
                    <button class="Searchbutton" onClick={() => this.Searchmusic()}>Search</button>
                </div>

                <fieldset class="track-info">
                    {this.state.data.map(sort_data => <article key={sort_data.trackId}>{sort_data.artistName}<br/>
                    {sort_data.trackName}<br/><img src={sort_data.artworkUrl100} alt={sort_data.trackId}/>
                    {console.log(sort_data)}<br/><audio controls><source src={sort_data.previewUrl}type='audio/mpeg'></source></audio>
                    <button class="favourite" onClick={() =>this.favouriteMusic(sort_data)}>Favorite</button>
                    </article>)} {/*Call the function created to search for the song*/}
                </fieldset>
            </div>
        )
    }
}

export default Music