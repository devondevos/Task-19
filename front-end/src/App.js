import React from 'react';

import './App.css';
import { HashRouter as Router, Switch, Route, Link } from "react-router-dom"


import Home from './components/Home'
import Music from './components/Music'
import Books from './components/Books'
import Favorite from './components/Favorite'

class App extends React.Component {
  cons
  render(){
    return (
      <Router >
        <Home/>
        <div id="links">
          <nav>
            <ul>
              <li>
                <Link to={'/'} class="homeNav">Home</Link>
              </li>
              <li>
                <Link to={'/Music'} class="homeNav">Search Music</Link>
              </li>
              <li>
                <Link to={'/Books'} class="homeNav">Search Books</Link>
              </li>
              <li>
                <Link to={'/Favorite'} class="homeNav">Favourites</Link>
              </li>
            </ul>
          </nav> {/*the navbar*/}
          <Switch>
          <Route path='/Home' component={Home}/>
            <Route path='/Music' component={Music}/>
            <Route path='/Books' component={Books}/>
            <Route path='/Favorite' component={Favorite}/>
          </Switch>
        </div> {/*link all the items in the navbar to the links*/}
      </Router>
    );
  }
}

export default App;
