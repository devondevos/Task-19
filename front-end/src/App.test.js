import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Music from './components/Music.jsx';
import renderer from 'react-test-renderer';

describe('Music component', () => {
	it('matches the snapshot', () => {
		const tree = renderer.create(<Music />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});