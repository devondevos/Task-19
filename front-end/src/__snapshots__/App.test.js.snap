// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music component matches the snapshot 1`] = `
<div
  id="music"
>
  <div
    class="searchItem"
  >
    <input
      class="Songinput"
      onChange={[Function]}
      placeholder="Search Tracks"
      type="text"
    />
    <button
      class="Searchbutton"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <fieldset
    class="track-info"
  >
     
  </fieldset>
</div>
`;
